packages/components/src/components/autocomplete/autocomplete.jsx

```diff
@@ -16,33 +16,135 @@ const IconArrow = ({ className, classNamePath }) => (
     </svg>
 );
 
+const KEY_CODE = {
+    ENTER  : 13,
+    ESCAPE : 27,
+    TAB    : 9,
+    KEYDOWN: 40,
+    KEYUP  : 38,
+};
+
+const not_found_default_text = 'No results found';
+
+const get_filtered_items = (val, list) => {
+    const is_string_array = list.length && typeof list[0] === 'string';
+
+    return list.filter(item => (
+        is_string_array ?
+            item.toLowerCase().includes(val)
+            : item.text.toLowerCase().includes(val)
+    ));
+};
 class Autocomplete extends React.PureComponent {
+    dropdown_ref     = React.createRef();
+    list_wrapper_ref = React.createRef();
+    list_item_ref    = React.createRef();
+
     state = {
         should_show_list: false,
-        filtered_items  : null,
+        filtered_items  : [... this.props.list_items],
         input_value     : '',
+        active_index    : null,
     };
 
     setInputWrapperRef = (node) => this.input_wrapper_ref = node;
 
-    // TODO: Prevent form submission when user is finished typing and keys in Enter
-    onKeyPressed = (event) => {
-        if (this.state.should_show_list) {
-            // keycode == 13 is Enter key
-            if (event.keyCode === 13) event.preventDefault();
+    onKeyPressed = event => {
+        const { active_index, filtered_items, should_show_list } = this.state;
+
+        switch (event.keyCode) {
+            case KEY_CODE.ENTER:
+                event.preventDefault();
+                this.hideDropdownList();
+                this.onItemSelection(filtered_items[active_index]);
+                break;
+            case KEY_CODE.TAB:
+                if (should_show_list) {
+                    this.hideDropdownList();
+                    this.onItemSelection(filtered_items[active_index]);
+                }
+                break;
+            case KEY_CODE.ESCAPE:
+                this.hideDropdownList();
+                break;
+            case KEY_CODE.KEYDOWN:
+                if (!should_show_list) this.showDropdownList();
+                this.setActiveDown();
+                break;
+            case KEY_CODE.KEYUP:
+                if (!should_show_list) this.showDropdownList();
+                else this.setActiveUp();
+                break;
+            default:
+                if (!should_show_list) this.showDropdownList();
+                break;
         }
     };
 
+    setActiveUp = () => {
+        const { active_index, filtered_items } = this.state;
+
+        if (typeof active_index === 'number') {
+            const next = active_index - 1;
+
+            if (next < 0) {
```

Comment:
what does next < 0 indicates, can you put a comment on it, or maybe put in into a variable with naming